flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobDetailsHandler.java

```diff
@@ -141,6 +142,11 @@ private static JobDetailsInfo createJobDetailsInfo(
                     executionState, jobVerticesPerState[executionState.ordinal()]);
         }
 
+        JobPlanInfo.RawJson streamGraphJson = null;
+        if (!StringUtils.isNullOrWhitespaceOnly(executionGraph.getStreamGraphJson())) {
```

Comment:
the return value of getStreamGraphJson will not be null.

Reply:
That's true, but it's safe to check non-null at the same time by `StringUtils.isNullOrWhitespaceOnly`? WDYT?

Reply:
I think isNullOrWhitespaceOnly will not throw an exception if the string is null.

If we want to perform a safety check, we should evaluate the state and throw an exception if it is not as expected.

So here maybe checkNotNull() is better.

Reply:
That's true, updated with `checkNotNull()`.